### ISSUE TITLE
Installation troubles

### ISSUE THREAD

**Issue by Plevuus** [GeorgeSG/philips_airpurifier_http#1]
I tried to install this custom component into my Home Assistant running on a Rpi 3B+. The first mentioned installation method failed, because I could not find this custom component in HACS. The second mentioned installation method also failed. The log reports `Platform error fan.philips_airpurifier - No module named 'pyairctrl'`. I tried `pip3 install py-air-control` from the CLI, but that didn't work either.

**Comment by GeorgeSG** [comment]
Hm. I'm not really sure how things stand with the manual installation and installing dependencies.

You should be able to install it via HACS though. Keep in mind, this is not a HACS default repo, so that's why you are not seeing it. 

You have to add it as a custom repository:

1. Go to integrations, open the menu in the top right, click "Custom Repositories": <img width="317" alt="image" src="https://user-images.githubusercontent.com/2514425/113562012-1884d880-960e-11eb-8bc5-dfa12802c3eb.png">
2. Paste the repository URL (`https://github.com/GeorgeSG/ha-philips-airpurifier/`) and select "Integration" as category: 
<img width="951" alt="image" src="https://user-images.githubusercontent.com/2514425/113562106-38b49780-960e-11eb-9602-cf656f4efbbb.png">

You should be able to see this repo in "Integrations" after that.

---

Also, remember to remove other forks / versions of the component you might have before installing this one. You should make sure you don't have a `philips-airpurifier` folder in your `custom_components` folder.

**Comment by Plevuus** [comment]
Thanks @GeorgeSG. Installation succeeded now! I missed the custom repro part. I should read better.

**Comment by GeorgeSG** [comment]
@Plevuus 👍 let me know if you have any other issues and feel free to open tickets with suggestions for improvements

**Comment by Plevuus** [comment]
Thanks again George.
I just installed the GitHub desktop app, created a local branch added the service to set the index used (IAI or PM2.5), and successfully tested it. I will try to upload that branch and try to create a pull request. I'm all new to this stuff, so it might be messy or incorrect.
The AC2889/10 seems to function a bit different than the other versions. Setting the speed doesn't work for now. For this model one has to set it to Manual mode and write the speed. Sleep and Turbo are also not modes but speeds. I also modified the  xMrVizzy version to get it fully working with the AC2889. I will probably do that again for myself. If you want, I'll also try to share that as a branch. 

EDIT: It turns out I cannot publish a branch, only a fork. Would a fork be the best way to share my modifications?

**Comment by GeorgeSG** [comment]
@Plevuus, yup, you can fork from here and open a PR or just share your branch.

Also, speeds are now deprecated for fans in Home Assistant. So I've dropped the `set_speed` support entirely here. Try using the new `set_percentage`. It's a little bit confusing, but basically, there's `set_preset_mode` for things like `allergy`, `night`, etc, and `set_percentage` for Speed 1, Speed 2, Speed 3 (which correspond to different percentages).

Here's how my fan entity looks:
<img width="401" alt="image" src="https://user-images.githubusercontent.com/2514425/113583268-1cc2ed00-9632-11eb-857f-72919791140e.png">

The "speed" slider (which is a percentage slider) has a couple of stops - 0% is off, 20% is silent, 40,60,80 are Speed 1, 2, 3, and 100% is Turbo.

Edit: if you're interested, check out the [Fan Developer Docs](https://developers.home-assistant.io/docs/core/entity/fan/) for more info

**Comment by Plevuus** [comment]
Hi George, Publishing the Branch doesn't work. I get this message:
![image](https://user-images.githubusercontent.com/46243037/113583810-66a7d500-962a-11eb-9ed6-02df4ab66610.png)
So, for now I forked.

I understand the changes in Home Assistant with respect to the speed changes. At least, I believe I do. But for some reason the API for setting the speed of the AC2889 seems to be a bit different.  I think line 331 of your code:
` await self._async_set_values({PHILIPS_SPEED: speed})`
Needs to change into something like this:
`await self._async_set_values({PHILIPS_MODE: 'M', PHILIPS_SPEED: philips_speed})`
for it to work with with the AC2889. At least I had to do something similar to xMrVizzy code to make it work. Setting the speed (percentage) for now doesn't have any effect.
I expect that also the sleep mode needs to move from the mode map to the speed map. I will do some experimenting later.

**Comment by GeorgeSG** [comment]
@Plevuus Hm, that kinda makes sense, though I feel like the purifier should automatically set itself in Manual Mode when the speed is changed (that's what mine does, but I guess it may be different).

Have you tried doing this via `py-air-control`'s CLI? I'm interested if it works by just setting the speed there, or you'll need to manually specify Manual mode as well.

I've been thinking that eventually we'll need per-device configuration. It looks like Philips aren't really consistent with their APIs and everything is slightly different on different devices. This may be the first one :)

**Comment by GeorgeSG** [comment]
With regards to speed, it looks like some devices have a "Sleep mode" and some devices have a "Silent Speed". I've added both. So if your device supports "Silent Speed", use `set_percentage` 20. If your device supports "Sleep mode", use `set_preset_mode` "sleep"

**Comment by Plevuus** [comment]
@GeorgeSG, Yes, I tested with the py-air-control CLI, but it is a long time ago I did so. I will re-try later. I'm afraid the different Philips models might behave sligtly different. My AC2889 has no "Silent Speed" nor a "Sleep Mode", but it has a "Sleep Speed".
Screenshot from the official 'Air Matters' App speed settings:
![image](https://user-images.githubusercontent.com/46243037/113771691-fe3d1e80-9723-11eb-84e9-b134c96d325a.png)
and the Mode settings:
![image](https://user-images.githubusercontent.com/46243037/113771631-ebc2e500-9723-11eb-9e60-698961095a15.png)
